07.ls_object/file_display.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+class FileDisplay
+  COLUMN_COUNT = 3
+
+  def initialize
+    @options = ARGV.getopts('alr').transform_keys(&:to_sym)
+    @file_details = fetch_file_details
+  end
+
+  def format_file_detail_strings
+    @options[:l] ? long_format : short_format
+  end
+
+  private
+
+  def fetch_file_details
+    flags = @options[:a] ? File::FNM_DOTMATCH : 0
+    file_names = Dir.glob('*', flags)
+    sorted_file_names = @options[:r] ? file_names.reverse : file_names
+    sorted_file_names.map { |file_name| FileDetail.new(file_name) }
+  end
+
+  def long_format
+    blocks_total = @file_details.map(&:blocks).sum
+    puts "total #{blocks_total}"
+
+    @file_details.each { |file_detail| puts format_file_detail(file_detail) }
+  end
+
+  def format_file_detail(file_detail)
+    max_size_map = build_max_size_map(@file_details)
+    stat_map = {
+      file_mode: file_detail.file_mode,
+      nlink: file_detail.nlink,
+      user_name: file_detail.user_name,
+      group_name: file_detail.group_name,
+      size: file_detail.size,
+      mtime: file_detail.mtime.strftime('%_m %_d %H:%M'),
+      name: file_detail.name
+    }
+    merged_file_detail = [
+      '%<file_mode>s',
+      "%<nlink>#{max_size_map[:nlink] + 1}s",
+      "%<user_name>-#{max_size_map[:user_name] + 1}s",
+      "%<group_name>-#{max_size_map[:group_name] + 1}s",
+      "%<size>#{max_size_map[:size]}s",
+      '%<mtime>2s',
+      "%<name>s\n"
+    ].join(' ')
+
+    format(merged_file_detail, stat_map)
```

Comment:
```suggestion
    [
      file_detail.file_mode,
      format("%#{max_size_map[:nlink] + 1}s", file_detail.nlink),
      # ...
    ].join(' ')
```

対応関係を確認するのに、stat_map と merged_file_detail の間で視線が上下してしまうので、わけずにまとめてしまった方がみやすいと思いました

Reply:
> 対応関係を確認するのに、stat_map と merged_file_detail の間で視線が上下してしまうので、わけずにまとめてしまった方がみやすいと思いました。

おっしゃる通り、分けずにまとめてしまったほうが読みやすいですし、読む時間もかからないかと思いました。
一つのメソッド内に`stat_map`と`merged_file_detail`を定義すれば問題ないと思っていましたが、結局のところ、対応関係を確認する作業はなくならないですよね😅
こちら修正しました！